File: .NET/Deviations/UpdateDeviation.cs
using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;

var boardsOnFireApiUrl = "YOUR URL"; // exempel: https://my-company.boardsonfireapp.com/api/v5/
var apiKey = "YOUR API KEY";
var deviationId = "YOUR DEVIATION ID";

using (var client = new HttpClient())
{
    client.BaseAddress = new Uri(boardsOnFireApiUrl);
    client.DefaultRequestHeaders.Accept.Clear();
    client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
    client.DefaultRequestHeaders.Add("x-api-key", apiKey);

    try
    {
        // Update field cause_description
        var requestBody = new
        {
            cause_description = "Something went wrong in production."
        };

        var jsonContent = JsonSerializer.Serialize(requestBody);
        var httpContent = new StringContent(jsonContent, Encoding.UTF8, "application/json");

        var response = await client.PatchAsync($"deviations/{deviationId}", httpContent);
        response.EnsureSuccessStatusCode();

        var data = await response.Content.ReadAsStringAsync();
        Console.WriteLine("Deviation:");
        Console.WriteLine(data);
    }
    catch (HttpRequestException e)
    {
        Console.WriteLine($"Request error: {e.Message}");
    }
}

File: Python/Deviations/UpdateDeviation.py
import requests
import json

boards_on_fire_api_url = "YOUR URL"  # example: https://my-company.boardsonfireapp.com/api/v5/
api_key = "YOUR API KEY"
deviation_id = "YOUR DEVIATION ID"

headers = {
    "Content-Type": "application/json",
    "x-api-key": api_key
}

# Update field cause_description
request_body = {
    "cause_description": "Something went wrong in production."
}

try:
    response = requests.patch(f"{boards_on_fire_api_url}/deviations/{deviation_id}", headers=headers, data=json.dumps(request_body))
    response.raise_for_status()
    data = response.json()
    print("Deviation:")
    print(data)
except requests.exceptions.RequestException as e:
    print(f"Request error: {e}")

